include/seqan3/io/sequence_file/format_fasta.hpp

```diff
@@ -269,7 +271,7 @@ class format_fasta
 
         if constexpr (!detail::decays_to_ignore_v<seq_type>)
         {
-            auto constexpr not_in_alph = !is_in_alphabet<seq_legal_alph_type>;
+            auto constexpr is_legal_alph = char_is_valid_for<seq_legal_alph_type>;
```

Comment:
Just out of curiosity, why the change from the not? Is that the better practice?

Reply:
**Technical reason:** The `char_is_valid_for` holds a function (with return type bool) and cannot be negated, while the `is_in_alphabet` was inherited from `char_predicate_base` and has the [not-operator](http://docs.seqan.de/seqan/3-master-dev/structseqan3_1_1detail_1_1char__predicate__base.html#aaa9a323dabf6386ebd0b0eb14a51d854) implemented. 

**Practical reason:** All other formats implement the `is_legal_alph` version and for consistency it should be changed. Additionally, the error message has been wrong here: _"Encountered an unexpected letter: is_in_alphabet\<seqan3::rna4\> evaluated to true on 'N'."_